### PR TITLE
🌱 Update the POST API request

### PR DESCRIPTION
**Pull request by azeemshaikh38**
PR updates the POST API request so that:

- It can use input `GITHUB_TOKEN`
- Takes the `branch` and `repo` inputs from the API or URL instead of header

### CODE REVIEW

**Review by laurentsimon — APPROVED** [review]

File: app/post_results.go [hunk 1]
```diff
@@ -174,84 +215,81 @@ func writeToBlobStore(ctx context.Context, bucketURL, filename string, data []by
...
 
 	cert := certs[0]
-
 	// Verify that cert isn't expired.
 	if err = cosign.CheckExpiry(cert, time.Unix(*entry.IntegratedTime, 0)); err != nil {
-		return "", "", "", "", fmt.Errorf("cosign certificate is expired: %v", err)
+		return nil, fmt.Errorf("error during cosign.CheckExpiry: %w", err)
 	}
+	return cert, nil
+}
```
[the remark this answers is not in the document]
Comment (azeemshaikh38) [reply app/post_results.go:254]: Sorry, how do you mean verify? Does extracting the info from the cert (see `extractCertInfo` fn) do what you are looking for?

[the remark this answers is not in the document]
Comment (laurentsimon) [reply app/post_results.go:254]: I don't think so. See what the slsa verifier does: https://github.com/slsa-framework/slsa-verifier/blob/main/pkg/provenance.go#L304
It does verify using the root CA (Fulcio). Maybe it'd one in another function...?
/cc @asraa to confirm

[the remark this answers is not in the document]
Comment (azeemshaikh38) [reply app/post_results.go:256]: I'm going to submit this to unblock myself from testing the Scorecard action and API integration. But will keep this conversation unresolved till we know the answer for sure.

[the remark this answers is not in the document]
Comment (laurentsimon) [reply app/post_results.go:256]: let's create an issue for it?

[the remark this answers is not in the document]
Comment (azeemshaikh38) [reply app/post_results.go:256]: Done.

**Pull request merged**